File: backend/StudioAEC.Api/Services/PlanningService.cs
using StudioAEC.Api.Models;

namespace StudioAEC.Api.Services;

public class PlanningService
{
    private readonly List<PlanningList> _lists =
    [
        new("backlog", "Backlog", 0),
        new("sprint", "Em andamento", 1),
        new("review", "Revisão", 2),
        new("done", "Concluído", 3)
    ];

    private readonly List<PlanningTask> _tasks =
    [
        new("t1", "Gravar módulo Revit Arquitetura — Fachadas", "sprint", "Alta", "Raniere", DateOnly.FromDateTime(DateTime.Today.AddDays(-5)), DateOnly.FromDateTime(DateTime.Today.AddDays(4)), 65, "sprint"),
        new("t2", "Atualizar biblioteca de famílias hidráulicas", "sprint", "Média", "Equipe BIM", DateOnly.FromDateTime(DateTime.Today.AddDays(-2)), DateOnly.FromDateTime(DateTime.Today.AddDays(8)), 40, "sprint"),
        new("t3", "Revisar template elétrico v3.2", "review", "Alta", "Marina", DateOnly.FromDateTime(DateTime.Today.AddDays(-10)), DateOnly.FromDateTime(DateTime.Today.AddDays(1)), 90, "review"),
        new("t4", "Planejar trilha Preventivo 2026", "backlog", "Baixa", "Raniere", DateOnly.FromDateTime(DateTime.Today.AddDays(3)), DateOnly.FromDateTime(DateTime.Today.AddDays(20)), 10, "backlog"),
        new("t5", "Publicar curso Climatização — Dutos", "done", "Média", "Lucas", DateOnly.FromDateTime(DateTime.Today.AddDays(-30)), DateOnly.FromDateTime(DateTime.Today.AddDays(-2)), 100, "done"),
        new("t6", "Compatibilização estrutural com arquitetura", "sprint", "Alta", "Ana", DateOnly.FromDateTime(DateTime.Today), DateOnly.FromDateTime(DateTime.Today.AddDays(12)), 25, "sprint"),
        new("t7", "Landing page novos treinamentos", "backlog", "Média", "Design", DateOnly.FromDateTime(DateTime.Today.AddDays(5)), DateOnly.FromDateTime(DateTime.Today.AddDays(15)), 0, "backlog")
    ];

    public IReadOnlyList<PlanningList> GetLists() => _lists;

    public IReadOnlyList<PlanningTask> GetTasks() => _tasks;

    public IReadOnlyList<WorkloadEntry> GetWorkload() =>
    [
        new("Raniere", 3, 32, 40),
        new("Marina", 2, 24, 40),
        new("Equipe BIM", 2, 28, 40),
        new("Lucas", 1, 16, 40),
        new("Ana", 2, 30, 40),
        new("Design", 1, 12, 40)
    ];

    public IReadOnlyList<GanttItem> GetGantt() =>
        _tasks.Select(t => new GanttItem(
            t.Id,
            t.Title,
            t.Assignee,
            t.StartDate,
            t.EndDate,
            t.Status switch
            {
                "done" => "#22c55e",
                "review" => "#f59e0b",
                "sprint" => "#6366f1",
                _ => "#94a3b8"
            })).ToList();
}

File: web/src/features/admin/services/planningService.ts
export interface PlanningListDto {
  id: string;
  name: string;
  order: number;
}

export interface PlanningTaskDto {
  id: string;
  title: string;
  status: string;
  priority: string;
  assignee: string;
  startDate: string;
  endDate: string;
  progressPercent: number;
  listId: string;
}

export interface WorkloadEntryDto {
  assignee: string;
  taskCount: number;
  hoursEstimated: number;
  capacityHours: number;
}

export interface GanttItemDto {
  id: string;
  title: string;
  assignee: string;
  start: string;
  end: string;
  color: string;
}

const LISTS: PlanningListDto[] = [
  { id: 'backlog', name: 'Backlog', order: 0 },
  { id: 'sprint', name: 'Em andamento', order: 1 },
  { id: 'review', name: 'Revisão', order: 2 },
  { id: 'done', name: 'Concluído', order: 3 },
];

const TASKS: PlanningTaskDto[] = [
  {
    id: 't1',
    title: 'Gravar módulo Revit Arquitetura — Fachadas',
    status: 'sprint',
    priority: 'Alta',
    assignee: 'Raniere',
    startDate: new Date(Date.now() - 5 * 86400000).toISOString().slice(0, 10),
    endDate: new Date(Date.now() + 4 * 86400000).toISOString().slice(0, 10),
    progressPercent: 65,
    listId: 'sprint',
  },
  {
    id: 't2',
    title: 'Atualizar biblioteca de famílias hidráulicas',
    status: 'sprint',
    priority: 'Média',
    assignee: 'Equipe BIM',
    startDate: new Date(Date.now() - 2 * 86400000).toISOString().slice(0, 10),
    endDate: new Date(Date.now() + 8 * 86400000).toISOString().slice(0, 10),
    progressPercent: 40,
    listId: 'sprint',
  },
  {
    id: 't3',
    title: 'Revisar template elétrico v3.2',
    status: 'review',
    priority: 'Alta',
    assignee: 'Marina',
    startDate: new Date(Date.now() - 10 * 86400000).toISOString().slice(0, 10),
    endDate: new Date(Date.now() + 1 * 86400000).toISOString().slice(0, 10),
    progressPercent: 90,
    listId: 'review',
  },
  {
    id: 't4',
    title: 'Planejar trilha Preventivo 2026',
    status: 'backlog',
    priority: 'Baixa',
    assignee: 'Raniere',
    startDate: new Date(Date.now() + 3 * 86400000).toISOString().slice(0, 10),
    endDate: new Date(Date.now() + 20 * 86400000).toISOString().slice(0, 10),
    progressPercent: 10,
    listId: 'backlog',
  },
  {
    id: 't5',
    title: 'Publicar curso Climatização — Dutos',
    status: 'done',
    priority: 'Média',
    assignee: 'Lucas',
    startDate: new Date(Date.now() - 30 * 86400000).toISOString().slice(0, 10),
    endDate: new Date(Date.now() - 2 * 86400000).toISOString().slice(0, 10),
    progressPercent: 100,
    listId: 'done',
  },
];

export function getPlanningLists(): PlanningListDto[] {
  return LISTS;
}

export function getPlanningTasks(): PlanningTaskDto[] {
  return TASKS;
}

export function getWorkload(): WorkloadEntryDto[] {
  return [
    { assignee: 'Raniere', taskCount: 3, hoursEstimated: 32, capacityHours: 40 },
    { assignee: 'Marina', taskCount: 2, hoursEstimated: 24, capacityHours: 40 },
    { assignee: 'Equipe BIM', taskCount: 2, hoursEstimated: 28, capacityHours: 40 },
    { assignee: 'Lucas', taskCount: 1, hoursEstimated: 16, capacityHours: 40 },
    { assignee: 'Ana', taskCount: 2, hoursEstimated: 30, capacityHours: 40 },
    { assignee: 'Design', taskCount: 1, hoursEstimated: 12, capacityHours: 40 },
  ];
}

export function getGantt(): GanttItemDto[] {
  const colors: Record<string, string> = {
    done: '#22c55e',
    review: '#f59e0b',
    sprint: '#6366f1',
  };
  return TASKS.map((t) => ({
    id: t.id,
    title: t.title,
    assignee: t.assignee,
    start: t.startDate,
    end: t.endDate,
    color: colors[t.status] ?? '#94a3b8',
  }));
}
